sycl/source/handler.cpp

```diff
@@ -845,11 +832,14 @@ event handler::finalize() {
 #endif
   }
 
+  // Because graph case is handled right above.
+  assert(Queue);
```

Comment:
Would it be better to use sycl::exception here? So, that we can catch exception even when asserts are disabled (release builds)